src/basic/efivars.c

```diff
@@ -77,58 +84,58 @@ int efi_get_variable(
                  * https://github.com/torvalds/linux/commit/bef3efbeb897b56867e271cdbc5f8adaacaeb9cd.
                  */
                 for (unsigned try = 0;; try++) {
-                        n = read(fd, &a, sizeof(a));
+                        n = read(fd, buf, (size_t) st.st_size);
```

Comment:
can you please read one byte more (and thus allocate one byte more above). and then if the returned size doesn't precisely match the expected size we should treat that as an error/concurrent change